EventFilter/HcalRawToDigi/plugins/alpaka/HcalDigisProducerPortable.cc

```diff
@@ -0,0 +1,198 @@
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiHostCollection.h"
+#include "DataFormats/HcalDigi/interface/alpaka/HcalDigiDeviceCollection.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class HcalDigisProducerPortable : public stream::EDProducer<> {
+  public:
+    explicit HcalDigisProducerPortable(edm::ParameterSet const& ps);
+    ~HcalDigisProducerPortable() override = default;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+  private:
+    void produce(device::Event&, device::EventSetup const&) override;
+
+  private:
+    // input product tokens
+    edm::EDGetTokenT<HBHEDigiCollection> hbheDigiToken_;
+    edm::EDGetTokenT<QIE11DigiCollection> qie11DigiToken_;
+
+    // type aliases
+    using HostCollectionPhase1 = hcal::Phase1DigiHostCollection;
+    using HostCollectionPhase0 = hcal::Phase0DigiHostCollection;
+
+    using DeviceCollectionPhase1 = hcal::Phase1DigiDeviceCollection;
+    using DeviceCollectionPhase0 = hcal::Phase0DigiDeviceCollection;
+
+    // output product tokens
+    device::EDPutToken<DeviceCollectionPhase1> digisF01HEToken_;
+    device::EDPutToken<DeviceCollectionPhase0> digisF5HBToken_;
+    device::EDPutToken<DeviceCollectionPhase1> digisF3HBToken_;
+
+    struct ConfigParameters {
+      uint32_t maxChannelsF01HE, maxChannelsF5HB, maxChannelsF3HB;
+    };
+    ConfigParameters config_;
+  };
+
+  void HcalDigisProducerPortable::fillDescriptions(edm::ConfigurationDescriptions& confDesc) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<edm::InputTag>("hbheDigisLabel", edm::InputTag("hcalDigis"));
+    desc.add<edm::InputTag>("qie11DigiLabel", edm::InputTag("hcalDigis"));
```

Comment:
@cms-sw/hcal-dpg-l2 do we need to have two different `InputTag` for these collections?
Aren't they guaranteed to always have the same name?

Reply:
`hbheDigisLabel ` label is to continue the support for Phase-0 HB, just like CUDA
`qie11DigiLabel` label is for Phase-1 HB and HE

Reply:
what `EDProducer` produces those two collections with different labels ?

Reply:
in other words, is there a concrete example where these two labels need to be different ?